main.go

```diff
@@ -133,18 +133,17 @@ func main() {
 				return nil
 			})
 		}
-		err := wg.Wait()
-		close(done)
-		if err != nil {
-			os.Exit(1)
-		}
+		done <- wg.Wait()
 	}()
 
 	for _, d := range flag.Args() {
 		walk(ch, d)
 	}
 	close(ch)
-	<-done
+	if err := <-done; err != nil {
+		fmt.Printf("error: %s\n", err)
```

Comment:
I think the error paths do a `log.Printf` already, and it seems weird to print the error to stdout.  Is this line needed?